### PR TITLE
github: workflows: add delete-branch, restart-argocd-app, switch-argocd-app workflows

### PR DESCRIPTION
**Pull request by paivin-dn**
## Motivation
We need to have shared workflows that delete branches, switch, and restart argocd apps.

## Solution
The `Commit and push changes to ${{ inputs.branch }}` step in the deploy workflow has been split into 2 stages:
- when we need to push changes into a new branch
- when we need to deploy changes to the current branch
Therefore, the `new_branch` parameter has been added to this workflow.

New workflows have been added:
- `delete-branch.yaml` - delete a branch in the target repository
- `switch-argocd-app.yaml` - install [argocd cli](https://argo-cd.readthedocs.io/en/stable/cli_installation/), connect to argocd instance and change targetRevision in an argocd app
- `restart-argocd-app.yaml` - install [argocd cli](https://argo-cd.readthedocs.io/en/stable/cli_installation/) and restart argocd app deployment

## Checklist

<details open=true>
  <summary>PR checklist (self-check before review)</summary>

- [X] The branch name fits the `pr-<prefix>-<task-name>` pattern;
- [X] The PR has a meaningful name using the `<prefix>: <description>` pattern;
- [X] The PR is properly labeled depending on the change;
- [X] The PR is assigned to you;
- [X] If the PR is based on another PR, links to them are added to the PR header and the "has PR dependency" label is set;
- [X] Self-review. Review your code, commits, and the PR description;
- [X] Make sure that you updated the corresponding docs in the `docs/` directory;
- [X] Links to the updated docs are added to the PR header (if any);
- [X] CI successfully finished;

</details>

### CODE REVIEW

File: .github/workflows/switch-argocd-app.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    steps:
+      - name: Install argocd cli
+        run: |
+          sudo curl -o /usr/bin/argocd -sSL https://github.com/argoproj/argo-cd/releases/latest/download/argocd-linux-amd64 && sudo chmod 755 /usr/bin/argocd
+      - name: Argocd cli login
+        run: |
+          argocd login argocd-server.argocd --username ${{ inputs.argocd_user }} --password ${{ secrets.argocd_user_password }} --insecure
+      - name: Switch the argocd app targetRevision branch (if available)
+        run: |
+          branch=`argocd app get dev-cloud-telemetry --output json | jq -r '.spec.source.targetRevision'`
```
Comment (glebignatieff) [inline .github/workflows/switch-argocd-app.yaml:52]: Nit: use the new nested command syntax – `$(..)` this one is [obsolete](https://wiki.bash-hackers.org/syntax/expansion/cmdsubst).

**Review by glebignatieff — COMMENTED** [review]
In which case do we need to restart a deployment? Is this how we run sync?

OK, I've read https://github.com/emlid/cloud-telemetry/pull/15 and now I have more context :)

**Review by glebignatieff — APPROVED** [review]
We've sorted things out with @paivin-dn :)

File: .github/workflows/switch-argocd-app.yaml (hunk 1 above)
Comment (paivin-dn) [reply .github/workflows/switch-argocd-app.yaml]: Fixed

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -26,6 +26,11 @@ on:
         description: "Service version"
         required: true
         type: string
+      new_branch:
+        description: "Use this parameter if you want to push changes to a new branch"
+        required: false
+        type: string
+        default: ''
```
Comment (iamsaywhat) [inline .github/workflows/deploy.yaml:33]: Can we set the default value to `${{ inputs.branch }}` and add `create_branch check: ${{ inputs.new_branch != ' ' }}`? Then we could have only one common step.

Comment (paivin-dn) [reply .github/workflows/deploy.yaml:33]: According to [documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputsinput_idtype):
```
on.workflow_call.inputs.<input_id>.type
Required if input is defined for the on.workflow_call keyword. The value of this parameter is a string specifying the data type of the input. This must be one of: boolean, number, or string.
```
We cannot use refers to other variables from the input.

**Review by anastasiia-garnaga — APPROVED** [review]
LGTM!

**Pull request merged**